pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -444,3 +515,50 @@ func (policy *regoEnforcer) ExtendDefaultMounts(mounts []oci.Mount) error {
 func (policy *regoEnforcer) EncodedSecurityPolicy() string {
 	return policy.base64policy
 }
+
+func (policy *regoEnforcer) EnforceExecInContainerPolicy(containerID string, argList []string, envList []string, workingDir string) error {
+	policy.mutex.Lock()
+	defer policy.mutex.Unlock()
+
+	input := map[string]interface{}{
+		"containerID": containerID,
+		"argList":     argList,
+		"envList":     envList,
+		"workingDir":  workingDir,
+	}
+
+	results, err := policy.query("exec_in_container", input)
+	if err != nil {
+		return err
+	}
+
+	allowed, err := policy.allowed("exec_in_container", results)
+	if err != nil {
+		return err
+	}
+
+	if allowed {
+		err = policy.updateMetadata(results)
+		if err != nil {
+			return fmt.Errorf("unable to update metadata: %w", err)
+		}
+		return nil
+	} else {
+		input_json, err := json.Marshal(input)
```

Comment:
looks like we might need a generic method that will query for the error.

Reply:
With the advent of metadata, most of the method specific code has been abstracted away. Definitely can add some generic methods.

Reply:
I've factored out the shared logic into a `enforce` method, removing most of the code duplication.